### PR TITLE
filter startableByUser

### PR DESCRIPTION
**Pull request by sirolf2009**
Move filter startableByUser to backend

### CODE REVIEW

**Comment by Klaas-Ritense** [comment]
It looks like there are still some unit tests failing

File: process-document/src/main/java/com/ritense/processdocument/service/ProcessDocumentAssociationService.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public interface ProcessDocumentAssociationService {
 
     ProcessDocumentDefinition getProcessDocumentDefinition(ProcessDefinitionKey processDefinitionKey, long documentDefinitionVersion);
 
-    List<? extends ProcessDocumentDefinition> findProcessDocumentDefinitions(String documentDefinitionName);
+    List<? extends ProcessDocumentDefinition> findProcessDocumentDefinitions(String documentDefinitionName, Boolean startableByUser);
```
Comment (theo-ritense) [inline process-document/src/main/java/com/ritense/processdocument/service/ProcessDocumentAssociationService.java]: This is a breaking change. This method should be a new one, where the old one calls it with `null`.

File: process-document/src/main/java/com/ritense/processdocument/repository/ProcessDocumentDefinitionRepository.java [hunk 2]
```diff
@@ -50,10 +50,11 @@ public interface ProcessDocumentDefinitionRepository extends
         "AND     pdd.processDocumentDefinitionId.documentDefinitionId.version = ( " +
         "   SELECT  MAX(dd.id.version) " +
         "   FROM    JsonSchemaDocumentDefinition dd " +
-        "   WHERE   dd.id.name = pdd.id.documentDefinitionId.name " +
-        ")")
-    List<CamundaProcessJsonSchemaDocumentDefinition> findAllByDocumentDefinitionNameAndLatestDocumentDefinitionVersion(
-        @Param("documentDefinitionName") String documentDefinitionName
+        "   WHERE   dd.id.name = pdd.id.documentDefinitionId.name " +        ") " +
+        "AND (:startableByUser IS NULL OR pdd.startableByUser = :startableByUser)")
+    List<CamundaProcessJsonSchemaDocumentDefinition> findAllByDocumentDefinitionNameAndLatestDocumentDefinitionVersionAndStartableByUser(
```
Comment (theo-ritense) [inline process-document/src/main/java/com/ritense/processdocument/repository/ProcessDocumentDefinitionRepository.java]: Why is this such an unholy method name when you're doing a custom query anyway?

File: process-document/src/main/java/com/ritense/processdocument/repository/ProcessDocumentDefinitionRepository.java [hunk 3]
```diff
@@ -50,10 +50,11 @@ public interface ProcessDocumentDefinitionRepository extends
...
+        @Param("documentDefinitionName") String documentDefinitionName,
+        @Param("startableByUser") Boolean startableByUser
```
Comment (theo-ritense) [inline process-document/src/main/java/com/ritense/processdocument/repository/ProcessDocumentDefinitionRepository.java]: Please annotate it with `@Nullable`

**Review by theo-ritense — APPROVED** [review]

**Pull request merged**